infrastructure/code/code.go

```diff
@@ -112,10 +115,55 @@ func (sc *Scanner) SupportedCommands() []snyk.CommandName {
 	return []snyk.CommandName{snyk.NavigateToRangeCommand}
 }
 
-func (sc *Scanner) Scan(ctx context.Context, _ string, folderPath string) (issues []snyk.Issue, err error) {
-	// When starting a scan for a folderPath that's already scanned, the new scan will wait for the previous scan
-	// to finish before starting.
-	// When there's already a scan waiting, the function returns immediately with empty results.
+func (sc *Scanner) Scan(ctx context.Context, path string, folderPath string) (issues []snyk.Issue, err error) {
+	if !sc.isSastEnabled() {
+		return issues, errors.New("SAST is not enabled")
+	}
+
+	sc.UpdateChangedFiles(path)
+
+	waiting, deferFn := sc.isScanWaiting(folderPath)
+	if waiting {
+		return []snyk.Issue{}, nil
+	}
+	defer deferFn()
+
+	// Start the scan
+	startTime := time.Now()
+	span := sc.BundleUploader.instrumentor.StartSpan(ctx, "code.ScanWorkspace")
+	defer sc.BundleUploader.instrumentor.Finish(span)
+
+	folderFiles, err := sc.files(folderPath)
+	if err != nil {
+		log.Warn().
+			Err(err).
+			Str("method", "domain.ide.workspace.Folder.ScanFolder").
+			Str("workspaceFolderPath", folderPath).
+			Msg("error getting workspace files")
+	}
+
+	metrics := sc.newMetrics(len(folderFiles), startTime)
+	results, err := sc.UploadAndAnalyze(span.Context(), folderFiles, folderPath, metrics)
+	if err != nil {
+		sc.changedFiles.ClearAll()
```

Comment:
Why clear in case of error? The opposite would seem more correct for me.

Reply:
Eagle eye! 👁️